### PR TITLE
[CDE - 824] - DashboardComponent: Make parameter propagation happen b…

### PR DESCRIPTION
**Pull request by eliofreitas**
…oth ways

### CODE REVIEW

**Comment by eliofreitas** [comment]
@pamval please review

**Comment by lucboudreau** [comment]
Error building command: /home/devuser/assembly-0.0.1/source/cde-base-699-1008bd0f-51e2-4369-b238-4cd8f26a4678/pom.xml (No such file or directory)

**Comment by pamval** [comment]
@afrjorge @jvelasques please review

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js [hunk 1]
```diff
@@ -78,9 +78,20 @@ define([
...
               var eventName = myParam + ":fireChange";
               var fun = function(evt) {
-                reqDash.fireChange(otherParam, evt.value);
+                if(reqDash.getParameterValue(otherParam) !== evt.value) {
+                  reqDash.fireChange(otherParam, evt.value);
+                }
               };
               myself.dashboard.on(eventName, fun);
+              var subDashboardEventName = otherParam + ":fireChange";
+              var subDashboardfun = function (evt) {
```
Comment (jvelasques) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: I'd like this to be consistent with the name chosen above, so `reqDashEventName` and `reqDashFun`

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js [hunk 2]
```diff
@@ -78,9 +78,20 @@ define([
...
+                if(myself.dashboard.getParameterValue(myParam) !== evt.value) {
+                  if(myself.oneWayMap === false) {
```
Comment (jvelasques) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: Was this intended in order to not activate this new behavior in already created dashboards? If not why not just `!myself.oneWayMap` ? Also this should be checked first, as it seems the "lighter" check.

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/component.xml [hunk 3]
```diff
@@ -64,6 +65,19 @@
             <Version>1.0</Version>
           </Header>
         </DesignerProperty>
+        <DesignerProperty>
+          <Header>
+            <Name>oneWayMap</Name>
+            <Parent>BaseProperty</Parent>
+            <DefaultValue>"false"</DefaultValue>
+            <Description>One Way Map</Description>
+            <Tooltip>If 'true' parameter propagation will only go in one direction</Tooltip>
```
Comment (jvelasques) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/component.xml]: Can we make it apparent which is the direction? Something like:
`If 'true' parameter propagation will only go from this Dashboard, to the required one`
@afrjorge comment on the above please

**Comment by eliofreitas** [comment]
@afrjorge @jvelasques updated

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js [hunk 4]
```diff
@@ -78,9 +78,18 @@ define([
             if(myself.isParameterPublic(otherParam)) {
               var eventName = myParam + ":fireChange";
               var fun = function(evt) {
-                reqDash.fireChange(otherParam, evt.value);
+                if(reqDash.getParameterValue(otherParam) !== evt.value) {
```
Comment (afrjorge) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: I think we should add a test to validate that **type conversion** doesn't happen.

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js [hunk 5]
```diff
@@ -78,9 +78,18 @@ define([
...
+                  reqDash.fireChange(otherParam, evt.value);
+                }
               };
               myself.dashboard.on(eventName, fun);
+              var reqDashEventName = otherParam + ":fireChange";
+              var reqDashFun = function (evt) {
+                if((myself.oneWayMap == false) && (myself.dashboard.getParameterValue(myParam) !== evt.value)) {
```
Comment (afrjorge) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: Also test this for **type conversion**.

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/component.xml (hunk 3 above)
Comment (afrjorge) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/component.xml]: I don't think we need the single quotes surrounding *true*. Checked the tooltips for the other custom components and single quotes seem to only be used for abbreviations.

**Comment by eliofreitas** [comment]
@jvelasques @afrjorge

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js (hunk 4 above)
Comment (eliofreitas) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: added

**Comment by jvelasques** [comment]
seems good, any other comment @afrjorge ?

**Comment by afrjorge** [comment]
@jvelasques no, also seems good

**Pull request merged**